homeassistant/components/emoncms_history.py

```diff
@@ -0,0 +1,92 @@
+"""
+A component which allows you to send data to Emoncms.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/emoncms_history/
+"""
+import logging
+from datetime import timedelta
+import json
+
+import voluptuous as vol
+import requests
+
+
+from homeassistant.const import (
+    CONF_API_KEY, CONF_WHITELIST,
+    EVENT_STATE_CHANGED, CONF_URL, STATE_UNKNOWN,
+    STATE_UNAVAILABLE)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import state as state_helper
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "emoncms_history"
+CONF_INPUTNODE = "inputnode"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_INPUTNODE): cv.positive_int,
+        vol.Required(CONF_WHITELIST, default=[]):
+            vol.All(cv.ensure_list, [cv.entity_id]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Setup the emoncms_history component."""
+    conf = config[DOMAIN]
+    whitelist = conf.get(CONF_WHITELIST)
+    json_body = {}
+
+    def emoncms_event_listener(event):
+        """Listen for new messages on the bus and sends them to Dweet.io."""
+        state = event.data.get('new_state')
+        if state is None or state.state in (STATE_UNKNOWN, "",
+                                            STATE_UNAVAILABLE) \
+                or state.entity_id not in whitelist:
+            return
+
+        try:
+            _state = state_helper.state_as_number(state)
+        except ValueError:
+            return
+
+        json_body[state.entity_id] = _state
+        json_str = json.dumps(json_body, separators=(",", ":"))
+        json_str = json_str.replace("\"", "")
+        json_str = json_str.replace("{", "").replace("}", "")
+        json_str = "{{{}}}".format(json_str)
+
+        send_data(conf.get(CONF_URL), conf.get(CONF_API_KEY),
+                  str(conf.get(CONF_INPUTNODE)), json_str)
+
+    hass.bus.listen(EVENT_STATE_CHANGED, emoncms_event_listener)
+
+    return True
+
+
+@Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
i initially knew about this and had put a warning in the documentation about this, but i removed it now as well as in the documentation. i only added the 1 second throttle to not hammer the emoncms site (in case it was the cloud version) but it has a negative impact. For reference the dweet.io history component had this same limitation